internal/engine/compiler/engine.go

```diff
@@ -557,39 +575,38 @@ func (e *engine) CompileModule(_ context.Context, module *wasm.Module, listeners
 			}
 		}
 
+		functionEndOffset := executableOffset + len(body)
+		if executableLength < functionEndOffset {
+			if executableLength == 0 {
+				executableLength = 65536
+			}
+			for executableLength < functionEndOffset {
+				executableLength *= 2
```

Comment:
curious if Go compiler is wise enough to compile this as left shift rather than multiplication as the left shift is much faster in general
```suggestion
				executableLength <<= 1
```

Reply:
I believe Go already performs those type of optimizations.